### ISSUE TITLE
Crash when exiting settings

### ISSUE THREAD

**Issue by nikobockerman** [TylerWilliamson/QuickWeather#49]
This might be related to #48 and to my inability to find any locations or to get any weather data. Anyway I faced a crash and got a stacktrace so reporting it here.

# Reproduction steps:
Initial state: API Key added and it's accepted by QuickWeather, no locations added.
1. Launch QuickWeather: opens settings as no locations are added
2. Click `>` to enter Locations page
3. Add a location
    * In my case I add a location through the map view (doesn't matter which but for example select some point inside Germany)  and give a random name like `ff`
4. Click `>` and `Finish` to enter main screen
    * At this point I only see a "loading" icon for a while after which I have only empty black page
5. Enter settings again
6. Click `>` to enter Locations page
7. Remove the `ff` location
    * List of locations is now empty
8. Use OS back gesture twice to exit from settings

# Actual result
QuickWeather crashes instead of showing home screen or closing the app

# Expected result
Some view to be shown or application to close. Anything but crashing.

# More information
App version: 2.2.6 from F-Droid
OS: Android 13
Phone: Pixel 6a

# Stack trace of the crash
```
type: crash
osVersion: google/bluejay/bluejay:13/TP1A.221005.003/2022101600:user/release-keys
package: com.ominous.quickweather:51
process: com.ominous.quickweather

java.lang.NullPointerException: Attempt to invoke interface method 'android.view.MenuItem android.view.MenuItem.setChecked(boolean)' on a null object reference
	at i3.b.d(SourceFile:671)
	at androidx.lifecycle.LiveData.b(Unknown Source:29)
	at androidx.lifecycle.LiveData.c(SourceFile:56)
	at androidx.lifecycle.LiveData.j(Unknown Source:14)
	at androidx.lifecycle.LiveData$a.run(Unknown Source:18)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7904)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ExecInit.main(ExecInit.java:49)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:355)

```

**Comment by TylerWilliamson** [comment]
It turns out that I already put in the fix I mentioned previously. The reason you had a crash was that the code happened on another thread, so I needed a similar fix in another place. That fix will be included in the next version

**Comment by TylerWilliamson** [comment]
Fixed in [2.2.7](https://github.com/TylerWilliamson/QuickWeather/releases/tag/2.2.7)
